samples/Kconfig.mcumgr_bt_ota_dfu

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config NCS_SAMPLE_MCUMGR_BT_OTA_DFU
+	bool "MCUmgr and OTA DFU over Bluetooth"
```

Comment:
This is no longer a thingy53 specific option, so it can be used on any nrf53 board - what happens if the user does not have a static PM config for their board and they're not using QSPI for secondary slots?

Reply:
Yes, the idea is to have the `NCS_SAMPLE_MCUMGR_BT_OTA_DFU` configuration option working for all targets (including nRF52 at some point in the future). This option will provide a configuration shortcut for users who want to quickly enable the MCUmgr BT OTA DFU solution in the default configuration. Currently, it is limited only to nRF53 devices and we can restrict it further to Thingy:53 if necessary.

We can impose additional limitations for unsupported configurations (like no QSPI for secondary slots). The user is required to change the default configuration if their specific use case does not match the defaults.

For the nRF5340 DK, the user is still required to provide MCUboot configuration, DTS definition of the external flash and partition layout. This configuration option is responsible for configuring mostly the MCUmgr part (DFU transport in the application image).

Reply:
Alternatively, I can define `pm_static`, external flash overlay and MCUboot configurations in the folder where we have the HCI RPMsg overlay currently ```bt_ota_dfu_overlay.conf```. If you think that there is a better place for such configurations than the ```subsys/mgmt/mcumgr``` folder, please let me know.

These additional files could make it work out of the box for the nRF5340 DK.

Reply:
Thingy:53 comes with predefined MCUboot bootloader configuration and PM static configuration. That makes enabling the feature way easier.

Reply:
I provided the support for both nRF5340 DK and Thingy53 that is compatible with their default MCUboot image configuration

Since the MCUboot image does not support Network Core updates on nRF5340 DK, you can only update the application image for this target. To enable OTA DFU for the nRF5340DK, it is enough to enable the following configurations:

```
CONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU=y
CONFIG_BOOTLOADER_MCUBOOT=y
```

See Kconfig diff for this change: https://github.com/nrfconnect/sdk-nrf/compare/fba65b675b73a9d57eb509a29ccfd64ccf8a4fcf..4784d606f81e20b49ecb4922352b5658b1b281b6